tests/rados/test_bluestore_data_compression.py

```diff
@@ -931,6 +949,239 @@ def validate_data_migration_between_pools():
+            \n compression_algorithm: {compression_algorithm} \
+            \n compression_required_ratio: {compression_required_ratio}---"
+            log.info(log_info_msg)
+
+            if (
+                compression_required_ratio is not None
+                and rados_obj.pool_inline_compression(
+                    pool_name=pool_name,
+                    compression_mode=compression_mode,
+                    compression_algorithm=compression_algorithm,
+                    compression_required_ratio=compression_required_ratio,
+                )
+                is False
+            ):
+                err_msg = f"Error enabling compression on pool : {pool_name}"
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+            log_info_msg = f"---3. Write IO to the pool {pool_name}---"
+            log.info(log_info_msg)
+            if not rados_obj.bench_write(
+                pool_name=pool_name,
+                max_objs=1,
+                byte_size=object_size,
+                num_threads=1,
+                type=type,
+                verify_stats=False,
+            ):
+                err_msg = f"Write IO failed on pool {pool_name}"
+                raise Exception(err_msg)
+
+            log_info_msg = "---4. Perform validations for bluestore data compression---"
+            log.info(log_info_msg)
+            log.info("(1) Validate compression is configured on pool")
+            validate_compression_configuration_on_pools(
+                rados_obj,
+                pool_name,
+                pool_configuration={
+                    "compression_required_ratio": compression_required_ratio,
+                    "compression_mode": compression_mode,
+                    "compression_algorithm": compression_algorithm,
+                    "compression_min_blob_size": None,
+                },
+            )
+
+            log.info(
+                "(2) Validating if compressed data < original data * compression_required_ratio"
+            )
+            pool_stats = get_pool_stats(rados_obj=rados_obj, pool_name=pool_name)
+            original_size = pool_stats["stored_data"]
+            compressed_data = pool_stats["compress_bytes_used"]
+
+            log_info_msg = f"""
+            \n Compressed data size = {compressed_data} \
+            \n Original data size = {original_size} \
+            \n compression_required_ratio = {compression_required_ratio} \
+            \n Original size * compression_required_ratio = {original_size * compression_required_ratio}\
+            \n compressed_size < original size * compression_required_ratio \
+            \n {compressed_data} < {original_size * compression_required_ratio}
+            \n Data should be compressed only if compressed_size < original size * compression_required_ratio.\
+            """
+            log.info(log_info_msg)
+
+            if not (compressed_data < original_size * compression_required_ratio):
```

Comment:
check could be <= ? as even if compressed_data == original_size * compression_required_ratio, it's still valid?